rfc/0006-threat-model.md

```diff
@@ -0,0 +1,262 @@
+in my cluster.
+
+# Detailed design
+[design]: #detailed-design
+
+The threat model of the Kubernetes admission control list 18 possible threats.
+All of them will be discussed sharing a brief description of it, possible mitigation
+and what we, as Kubewarden team, should do about it (if necessary). As well as
+help our users to be aware of the risks.
+
+If more details about each threat is needed, go to the original document here:
+https://github.com/kubernetes/sig-security/tree/main/sig-security-docs/papers/admission-control
+
+
+## Threat 1 - Attacker floods webhook with traffic preventing its
+
+An attacker who has access to the Webhook endpoint, at the network level, could
+send large  quantities of traffic, causing an effective denial of service to the
+admission controller.
+
+**Mitigation**
+
+Webhook fails closed. In other words, if the webhook does not respond in time,
+for any reasion, API server should reject the request. We are safe on this.
+Kubewarden default behavior  already does that.
+
+## Threat 2 - Attacker passes workloads which require complex  processing causing timeouts
+
+An attacker, who can access the admission controller at a network level, passes
+requests to  the admission controller which require complex processing, causing
+timeouts as the  admission controller uses compute power to process the workloads
+
+**Mitigation**
+Webhook fail closed and authenticate callers. We are safe, Kubewarden already does that.
+
+## Threat 3 - Attacker exploits misconfiguration of webhook to bypass
+An attacker, who has rights to create workloads in the cluster, is able to exploit
+a mis-configuration to bypass the intended security control
+
+**Mitigation**
+Regular reviews of webhook configuration catch issues.
+
+**To do**
+We may provide some mechanism to notify operators when a webhook is changed.
+Maybe the controller can watch its webhooks, fix when the configuration is
+changed and notify the operator.
+
+### Threat 4 - Attacker has rights to delete or modify the k8s webhook  object
+
+An attacker who has Kubernetes API access, has sufficient privileges to delete
+the webhook  object in the cluster.
+
+**Mitigation**
+RBAC rights are strictly controlled.
+
+**To do**
+I think most of the RBAC is not Kubewarden responsibility. But we can help our users if we:
+- Warning them in our docs and *suggest* some minimum RBAC to be used.
+- Provide a policy which detect RBAC changes and **maybe** block them.
+
+
+## Threat 5 - Attacker gets access to valid credentials for the webhook
+An attacker gains access to valid client credentials for the admission controller webhook
+
+**Mitigation**
+Webhook fails closed.
+
+Kubewarden is failed closed. Thus, we should be fine.
+
+## Threat  6 - Attacker gains access to a cluster admin credential
+
+An attacker gains access to a cluster-admin level credential in the kubernetes cluster.
+
+**Mitigation**
+N/A
+
+I cannot see how Kubewarden can help here.
```

Comment:
It is also not very clear in the original description "via a number of mechanisms.".